steamclog/src/main/java/com/steamclock/steamclog/Destinations.kt

```diff
@@ -186,38 +204,44 @@ internal class ExternalLogFileDestination : Timber.DebugTree() {
         return logDirectory
     }
 
-    fun getExternalFile(): File? {
+    /**
+     * Will get the external file we are currently writing to.
+     */
+    private fun getExternalFile(): File? {
         val expiryMs = SteamcLog.config.autoRotateConfig.fileRotationSeconds * 1000 // defaults to 10 minutes
 
         if (isCachedLogFileValid(expiryMs)) {
             return cachedCurrentLogFile
         }
 
-        val currentFile = findAvailableLogFile(expiryMs)
-        updateCachedLogFile(currentFile)
-        return currentFile
+        // Find new available log file and update cached variables
+        cachedCurrentLogFile = findAvailableLogFile(expiryMs)
+        logFileCreatedTimeEstimate = Date().time
```

Comment:
what kind of problems would using `cachedCurrentLogFile.lastModifiedTime()` cause

Reply:
Also, this seems to imply that all 10 log files will have the same `logFileCreatedTimeEstimate` even though each logfile could be selected at different times, is that expected behaviour

Reply:
> Also, this seems to imply that all 10 log files will have the same logFileCreatedTimeEstimate even though each logfile could be selected at different times, is that expected behaviour

Yes, technically any of the files could be selected to be used as the next available file (although the logic attempts to pick the oldest one), so in theory, yes that was the expected behaviour when the code was first written I believe.

With the changes I have made in your first comment though, this will no longer be the case. Now a selected file should have a `logFileCreatedTimeEstimate` of the last modified time so they would not all be the same.